### ISSUE TITLE
Faiss throws up TypedStorage is Deprecated UserWarning

### ISSUE THREAD

**Issue by chichieinstein** [KevinMusgrave/pytorch-metric-learning#618]
I am trying to train a transformer neural network using soft triplet loss. I upgraded PyTorch to 2.0 recently, and also installed the latest version of pytorch-metric-learning.

At the end of a training iteration, when accuracies are calculated, I am getting the TypedStorage is Deprecated Warning. Is this expected behavior? Should I turn this warning off inside the faiss package?

**Comment by chichieinstein** [comment]
![Screenshot from 2023-04-24 18-06-13](https://user-images.githubusercontent.com/93373310/234150022-e9e5504c-d784-4901-ac21-a33f650fe4c4.png)

This is a screen-shot of the Warning.

**Comment by chichieinstein** [comment]
To add to this, it was very difficult to make the latest PyTorch version 2.0 work with PML and Faiss. I was getting a lot of deprecation warnings. So, just an added question : is there a specific PyTorch version that I should use with this library. The README seems to suggest any version of Torch >= 1.6 would work.

**Comment by KevinMusgrave** [comment]
Maybe related: https://github.com/facebookresearch/faiss/issues/2817

**Comment by chichieinstein** [comment]
Yes. I did not modify the default behavior of the train with classifier class (yet). All issues I have had are with the Faiss package.

**Comment by chichieinstein** [comment]
I have swept this warning under the carpet by ignoring the UserWarning. And it works.
